src/Pulsar.Client/Internal/ClientCnx.fs

```diff
@@ -123,73 +125,68 @@ type ClientCnx (broker: Broker,
-                        let messageCheckSum  = reader.ReadInt32() |> int32FromBigEndian
-                        let metadataPointer = stream.Position
-                        let metatada = Serializer.DeserializeWithLengthPrefix<MessageMetadata>(stream, PrefixStyle.Fixed32BigEndian)
-                        let payloadPointer = stream.Position
-                        let metadataLength = payloadPointer - metadataPointer |> int
-                        let payloadLength = frameLength - (int payloadPointer)
-                        let payload = reader.ReadBytes(payloadLength)
-                        stream.Seek(metadataPointer, SeekOrigin.Begin) |> ignore
-                        CRC32C.Get(0u, stream, metadataLength + payloadLength) |> int32 |> invalidArgIf ((<>) messageCheckSum) "Invalid checksum" |> ignore
-                        XCommandMessage ((command.Message, metatada, payload), buffer.GetPosition(int64 frameLength))
-                    | BaseCommand.Type.LookupResponse ->
-                        XCommandLookupTopicResponse (command.lookupTopicResponse, buffer.GetPosition(int64 frameLength))
-                    | BaseCommand.Type.Ping ->
-                        XCommandPing (command.Ping, buffer.GetPosition(int64 frameLength))
-                    | BaseCommand.Type.ProducerSuccess ->
-                        XCommandProducerSuccess (command.ProducerSuccess, buffer.GetPosition(int64 frameLength))
-                    | BaseCommand.Type.Success ->
-                        XCommandSuccess (command.Success, buffer.GetPosition(int64 frameLength))
-                    | BaseCommand.Type.SendError ->
-                        XCommandSendError (command.SendError, buffer.GetPosition(int64 frameLength))
-                    | BaseCommand.Type.CloseProducer ->
-                        XCommandCloseProducer (command.CloseProducer, buffer.GetPosition(int64 frameLength))
-                    | BaseCommand.Type.CloseConsumer ->
-                        XCommandCloseConsumer (command.CloseConsumer, buffer.GetPosition(int64 frameLength))
-                    | BaseCommand.Type.ReachedEndOfTopic ->
-                        XCommandReachedEndOfTopic (command.reachedEndOfTopic, buffer.GetPosition(int64 frameLength))
-                    | BaseCommand.Type.GetTopicsOfNamespaceResponse ->
-                        XCommandGetTopicsOfNamespaceResponse (command.getTopicsOfNamespaceResponse, buffer.GetPosition(int64 frameLength))
-                    | BaseCommand.Type.Error ->
-                        XCommandError (command.Error, buffer.GetPosition(int64 frameLength))
-                    | _ as unknownCommandType ->
-                        InvalidCommand (Exception(sprintf "Unknown command type: '%A'" unknownCommandType))
-                with
-                | ex ->
-                    InvalidCommand ex
+            if (array.Length <= frameLength)
```

Comment:
= case forgotten

Reply:
There's another now. But thats exactly the case where you probably don't need an exception?

Reply:
My comment was about the comparison line, not about exception =)